src/main/java/org/gfccollective/vavr/future/ScalaFutureAdapter.java

```diff
@@ -0,0 +1,92 @@
+package org.gfccollective.vavr.future;
+
+import org.gfccollective.vavr.VavrConverters.VavrTryConverter;
+import org.gfccollective.vavr.VavrConverters.VavrOptionConverter;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Consumer;
+
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+import scala.concurrent.Await;
+import scala.concurrent.Future;
+import scala.concurrent.duration.Duration;
+
+/**
+ * ScalaFutureAdapter is implemented in Java because it
+ * is impossible to implement in Scala. The Scala compiler
+ * does not allow us to subclass io.vavr.concurrent.Future.
+ * Vavr's Future implements a method called 'eq' and this
+ * makes the Scala compiler very unhappy.
+ */
+public class ScalaFutureAdapter<T>
+    implements io.vavr.concurrent.Future<T> {
+
+    private final Future<T> scalaFuture;
+
+    public ScalaFutureAdapter(Future<T> scalaFuture) {
+        this.scalaFuture = scalaFuture;
+    }
+
+    protected Future<T> getScalaFuture() {
+        return scalaFuture;
+    }
+
+    @Override
+    public io.vavr.concurrent.Future<T> await() {
+        return await(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    public io.vavr.concurrent.Future<T> await(final long timeout, final TimeUnit unit) {
+        if (!scalaFuture.isCompleted()) {
+            try {
+                Await.result(scalaFuture, Duration.apply(timeout, unit));
```

Comment:
this is a blocking operation..

Reply:
> this is a blocking operation..

Yes.  

<img width="1200" alt="vavr-concurrent-Future-await" src="https://user-images.githubusercontent.com/30938/93727458-1e363780-fb70-11ea-9a56-56bf5fe7a97d.png">